Gems/Multiplayer/Code/Source/MultiplayerStats.cpp

```diff
@@ -253,4 +257,40 @@ namespace Multiplayer
         handlers.m_rpcSent.Connect(m_events.m_rpcSent);
         handlers.m_rpcReceived.Connect(m_events.m_rpcReceived);
     }
+
+    void MultiplayerStats::RecordFrameTime(AZ::TimeUs networkFrameTime)
+    {
+        m_framesSinceLastMetric++;
+        m_accumulatedNetworkTimeSinceLastMetric += networkFrameTime;
+    }
+
+    void MultiplayerStats::RecordMetrics()
+    {
+        if (const auto* eventLoggerFactory = AZ::Interface<AZ::Metrics::IEventLoggerFactory>::Get())
+        {
+            if (auto* eventLogger = eventLoggerFactory->FindEventLogger(NetworkingMetricsId))
+            {
+                using EventObjectStorage = AZStd::fixed_vector<AZ::Metrics::EventField, 8>;
+
+                EventObjectStorage argsContainer;
+                const uint64_t averageFrameTime = m_framesSinceLastMetric > 0 ?
+                    static_cast<uint64_t>(m_accumulatedNetworkTimeSinceLastMetric) / m_framesSinceLastMetric : 0;
+                argsContainer.emplace_back("AverageNetworkFrameTimeUs", averageFrameTime);
+                argsContainer.emplace_back("NumEntities", m_entityCount);
+
+                m_framesSinceLastMetric = 0;
+                m_accumulatedNetworkTimeSinceLastMetric = AZ::Time::ZeroTimeUs;
+
+                AZ::Metrics::InstantArgs instantArgs;
+                instantArgs.m_name = "Metrics Event";
+                instantArgs.m_cat = "Networking";
+                instantArgs.m_args = argsContainer;
+                instantArgs.m_scope = AZ::Metrics::InstantEventScope::Process;
+
+                eventLogger->RecordInstantEvent(instantArgs);
```

Comment:
This seems more like a Counter Event in according to the Google Trace Event Format [Document](https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/edit#heading=h.msg3086636uq)